envi/__init__.py

```diff
@@ -44,39 +44,59 @@
 
 ARCH_MASK        = 0xffff0000   # Masked into IF_FOO and BR_FOO values
 
+# Code Maturity Ratings.  Also used for Unittest categories
+MAT_NONE = 0            # does not exist
+MAT_ALPHA = 1           # code exists, but immature.  no promises
+MAT_BETA = 2            # kinda works, not overly confident
+MAT_TESTING = 3         # works ok, initial RC
+MAT_PROD = 4            # production
+
+maturity = {v:k for k,v in globals().items() if k.startswith('MAT_')}
```

Comment:
I really don't like this pattern. Both the lowercase for a top level export as well as the conceptual split between things like `MAT_PROD` being a constant you can grab and `maturity` being a dictionary.

Reply:
sorry, i don't follow at all.  can you explain it again?

Reply:
sorry, if there's a better way to do reverse lookups for constants, i'm happy to entertain a better way.

i'm confused about your comment here, tho.  
this is *specifically* so that you can give a maturity level name lookup from a constant number.

still, this will go away if we can agree on how to do semver here.